### PR TITLE
fixed jet corrections

### PR DESCRIPTION
**Pull request by jpata**
Jet corrections only ran once, also works in case one does not recorrect jets.

@arizzi this is to fix https://github.com/cbernet/cmssw/pull/41#issuecomment-121870472

### CODE REVIEW

File: PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py [hunk 1]
```diff
@@ -100,13 +100,22 @@ def process(self, event):
...
+            #apply nominal corrections
+            for jet in allJets:
+                corr = jet.corr
+                if corr > 0:
+                    jet.setP4(jet.p4() * (corr * jet.rawFactor()))
+                    jet.setRawFactor(1.0/corr)
+                else:
+                    badjets += [jet]
+            for jet in badjets:
+                allJets.remove(jet)
```
Comment (arizzi) [inline PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py]: mhh removing jets doesn't look a good idea... I'd rather flag them

**Comment by jpata** [comment]
OK, I removed the removal (originally after JetReCalibrator.correctAll: https://github.com/vhbb/cmssw/blob/vhbbHeppy74X/PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py#L41)

Now we can recognize bad jets by corr<=0.

**Comment by arizzi** [comment]
ah... if it was already there... then fine I thought it was a new feature!

**Comment by arizzi** [comment]
why did you close this one? is there a new PR in its place?

**Comment by jpata** [comment]
I closed by accident when cleaning up branches in my fork. Restoring.

**Comment by arizzi** [comment]
thanks, reading the thread it seems that we could revert to jet-removal rather than jet-flagging as this is the standard procedure in the recalibrator, isn't it?

**Comment by jpata** [comment]
Yes, I'll update this PR today.

**Comment by jpata** [comment]
this should be OK now. jet corrections are optional, bad jets are removed as per JetRecalibrator.

**Comment by cbernet** [comment]
Hi Joosep, 

The new heppy branch is now 
https://github.com/cbernet/cmssw/tree/heppy_7_4_7
Could you please PR there instead? 

Thanks, 

Colin

**Pull request closed without merging**